src-theme/src/routes/clickgui/setting/inventoryPreset/config/PresetItemSelector.svelte

```diff
@@ -0,0 +1,312 @@
+    import {REST_BASE} from "../../../../../integration/host";
+    import PresetTooltip from "./PresetTooltip.svelte";
+
+    export let setItem: (item: PresetItem) => void
+    export let filter: ((item: string) => boolean) | null = null
+
+    export let canSelectAny: boolean
+    export let canSelectIgnore: boolean
+
+    const commonItems: PresetItem[] = [
+        {
+            type: "SINGLE",
+            item: "minecraft:shield"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:ender_pearl"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:arrow"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:bow"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:crossbow"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:fishing_rod"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:golden_apple"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:enchanted_golden_apple"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:water_bucket"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:lava_bucket"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:flint_and_steel"
+        },
+        {
+            type: "SINGLE",
+            item: "minecraft:shears"
+        },
+    ];
+
+    interface GenericPresetItemList {
+        item: PresetItem
+        name: string,
+        tooltip?: string | null
+    }
+
+    const genericItems: GenericPresetItemList[] = [
+        {
+            item: { type: "GROUP", group: "WEAPON" },
+            name: "Weapon"
+        },
+        {
+            item: { type: "GROUP", group: "FOOD" },
+            name: "Food"
+        },
+        {
+            item: { type: "GROUP", group: "BLOCK" },
+            name: "Blocks"
+        },
+        {
+            item: { type: "GROUP", group: "AXE" },
+            name: "Axe"
+        },
+        {
+            item: { type: "GROUP", group: "PICKAXE" },
+            name: "Pickaxe"
+        },
+        {
+            item: { type: "GROUP", group: "HOE" },
+            name: "Hoe"
+        },
+        {
+            item: { type: "GROUP", group: "POTION" },
+            name: "Potions"
+        },
+        {
+            item: { type: "GROUP", group: "THROWABLE" },
+            name: "Throwables"
+        },
+        {
+            item: { type: "GROUP", group: "ARROWS" },
+            name: "Arrows"
+        },
+        ...(canSelectAny ? [{
+            item: { type: "ANY" } as const,
+            name: "Any",
+            tooltip: "This slot is simply skipped during filtering..."
+        }] : []),
+        ...(canSelectIgnore ? [{
+            item: { type: "IGNORE" } as const,
+            name: "Ignore",
+            tooltip: "There can be nothing in this slot"
```

Comment:
Ignore means that the InvCleaner may not touch anything which is in this slot (even if it considers it useless)